test-data/preprocessor/test/CounterTest.sol

```diff
@@ -0,0 +1,25 @@
+import {Counter} from "src/Counter.sol";
+import {Counter as CounterV1} from "src/v1/Counter.sol";
+import "src/CounterB.sol";
+import "src/CounterC.sol";
+import "src/CounterD.sol";
+
+contract CounterTest {
+    Counter public counter;
+    Counter public counter2 = new Counter();
+    CounterB public counter3 = new CounterB(address(this), 44, true, address(this));
+    CounterV1 public counterv1;
+
+    function setUp() public {
+        counter = new Counter();
+        counterv1 = new CounterV1(     );
+        type(CounterV1).creationCode;
+        CounterB counterB = new CounterB(address(this), 15,           false, address(counter));
```

Comment:
can we add a test for named arguments `{a:address(this), ..}` and call options `{salt:123}`

Reply:
added in https://github.com/foundry-rs/compilers/pull/252/commits/64c5ecdc8572b6eef410e3385d29dbb0f61c7558 with additional logic to handle combination of both, e.g.
```Solidity
CounterB public counter6 = new CounterB {salt: bytes32("123")}   (address(this), 44, true, address(this));
```
however it fails this format
```Solidity
CounterB public counter6 = new CounterB {salt: bytes32("123")}   (     address(this), 44, true, address(this)   );
```

@DaniPopes could you pls check the logic and suggest if better way to handle such?

Reply:
this would override the `{salt...}` right? I guess it's fine but technically incorrect since it will ignore the argument

Reply:
yes, that's right. not sure, would this work if we add it as a regular arg and pass it to constructor?

Reply:
would need foundry support for another overload of `deployCode` with `bytes32 salt` that uses create2 scheme instead of create: https://github.com/foundry-rs/foundry/blob/e0f87ad31d929abcf7f0eb96952e1805dc9d2c22/crates/cheatcodes/src/fs.rs#L328

Reply:
kk, makes sense, will add such support

Reply:
actually I think we need more variations for msg.sender as well, pls check https://github.com/foundry-rs/foundry/pull/10178